### PR TITLE
Fixes issue with crashing linter rule, adds missing unit tests

### PR DESCRIPTION
**Pull request by schuylermartin45**
- Streamlines the auto-fix tests by using a test fixture setup for all auto fix tests
- Adds our first multi-output unit test to the auto fix roster
- Fixes a linting rule that was known to crash using some percy enhancements

### CODE REVIEW

**Comment by schuylermartin45** [comment]
The failure is, ironically, caused by the new WARNINGs error code. Exit code of `100` should be ignored by that GitHub action (as we are purposefully publishing this package as `noarch`)

File: tests/conftest.py [hunk 1]
```diff
@@ -150,6 +156,14 @@ def assert_on_auto_fix(check_name: str, arch: str = "linux-64") -> None:
...
 
-    assert len(messages) == 1
-    assert messages[0].auto_fix_state == AutoFixState.FIX_PASSED
+    # Ensure that:
+    #   - The rule triggered the expected number of times
+    #   - That the correct rule triggered
+    #   - And that the rule was actually fixed
+    # This supports
+    assert len(messages) == num_occurrences
+    for i in range(num_occurrences):
```
Comment (schuylermartin45) [inline tests/conftest.py:165]: I'm not usually a fan of having loops in a unit test, but I think it makes sense in this scenario.

File: anaconda_linter/lint/check_syntax.py [hunk 2]
```diff
@@ -20,45 +23,48 @@ class version_constraints_missing_whitespace(LintCheck):
 
     """
 
+    # TODO Future: percy should eventually have support for constraints. This regex may not be
+    # sufficient enough to catch all accepted formats
```
Comment (JeanChristopheMorinPerso) [inline anaconda_linter/lint/check_syntax.py:27]: Supported symbols are documented at https://docs.conda.io/projects/conda-build/en/stable/resources/package-spec.html#build-version-spec.

File: anaconda_linter/lint/check_syntax.py [hunk 3]
```diff
@@ -20,45 +23,48 @@ class version_constraints_missing_whitespace(LintCheck):
...
-            for n, spec in enumerate(recipe.get(path, [])):
-                has_constraints = constraints.search(spec)
-                if has_constraints:
-                    # The second condition is a fallback.
-                    # See: https://github.com/anaconda-distribution/anaconda-linter/issues/113
-                    space_separated = has_constraints[1].endswith(" ") or " " in has_constraints[0]
-                    if not space_separated:
-                        self.message(section=f"{path}/{n}", data=True, output=output)
+            output: Final[int] = -1 if not path.startswith("/outputs") else int(path.split("/")[2])
+            n = int(path.split("/")[-1])
```
Comment (JeanChristopheMorinPerso) [inline anaconda_linter/lint/check_syntax.py:40]: This this necessary? It doesn't seems to be used.

File: anaconda_linter/lint/check_syntax.py [hunk 4]
```diff
@@ -20,45 +23,48 @@ class version_constraints_missing_whitespace(LintCheck):
...
-                has_constraints = constraints.search(spec)
-                if has_constraints:
-                    # The second condition is a fallback.
-                    # See: https://github.com/anaconda-distribution/anaconda-linter/issues/113
-                    space_separated = has_constraints[1].endswith(" ") or " " in has_constraints[0]
-                    if not space_separated:
-                        dep, ver = has_constraints.groups()
-                        self.recipe.replace(spec, f"{dep} {ver}", within="requirements")
-        return True
+        def _rm_whitespace(parser: RecipeParser):
```
Comment (JeanChristopheMorinPerso) [inline anaconda_linter/lint/check_syntax.py:62]: Add instead of rm?

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: anaconda_linter/lint/check_syntax.py (hunk 3 above)
Comment (schuylermartin45) [reply anaconda_linter/lint/check_syntax.py:40]: Good catch! This is a holdover from the original logic. Pylint would have caught it, but it's being used by the messaging system in error.

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -62,20 +62,3 @@ jobs:
             source $CONDA/bin/activate
             conda install conda-build
```
Comment (schuylermartin45) [inline .github/workflows/ci.yaml:63]:
```suggestion
            testing
```

Comment (schuylermartin45) [reply .github/workflows/ci.yaml:63]: ^ Ignore, I'm testing how the commit suggestion system works.

**Pull request merged**